packages/core/src/utils.ts

```diff
@@ -60,19 +60,7 @@ export function isResourceWithId<T extends Resource>(
   resource: unknown,
   resourceType?: T['resourceType']
 ): resource is WithId<T> {
-  // Resource must be an object
-  if (!resource || typeof resource !== 'object') {
-    return false;
-  }
-  // Must contain `resourceType` and `id` keys
-  if (!('resourceType' in resource && 'id' in resource)) {
-    return false;
-  }
-  // Optionally check resource type
-  if (resourceType && resource.resourceType !== resourceType) {
-    return false;
-  }
-  return true;
+  return isResource(resource, resourceType) && 'id' in resource;
```

Comment:
```suggestion
  return isResource(resource, resourceType) && 'id' in resource && typeof resource.id === 'string';
```
Probably want to check typeof on `resource.id` since `in` only checks that the field exists on the object and not that it is defined
<img width="150" alt="Screenshot 2024-12-06 at 10 26 23 AM" src="https://github.com/user-attachments/assets/720ba77f-a37b-4f7d-9fdf-819463477e26">